### PR TITLE
Sjohn/bleu

### PR DESCRIPTION
**Pull request by sjohn4**
Adds Bleu evaluation and changes the core evaluation to allow for model.generate of responses instead of reading logits and picking the highest score which usually give to nonsensical sentences.  DO NOT MERGE THIS. You can take a quick look.

### CODE REVIEW

**Comment by sjohn4** [comment]
This Pr should be done differently, this will not work well.
